tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,314 @@
+"""
+==========================
+Arranging Axes in a Figure
+==========================
+
+Often more than one axes is wanted on a figure at a time, and usually
+we want to organize those axes into a regular grid.  Matplotlib has a
+variety of tools for working with grids of axes that have evolved over
+the history of the library.  Here we will discuss the tools we think
+users should use most often, and then dicuss some of the older tools,
+and the tools that underpin how axes are organized.
+
+How to create grid-shaped combinations of axes:
+
+`~matplotlib.pyplot.subplots`
+    The primary function used to create figures and a grid of axes.  It is
+    similar to `.pyplot.subplot`, but creates and places all axes on the
+    figure at once, and returns an object array with handles for
+    the axes in the grid.  See also `.Figure.subplots`.
+
+or
+
+`~matplotlib.pyplot.subplot_mosaic`
+    A more flexible function used to create figures and a grid of axes,
+    with the added flexibility that some of the axes can span rows or
+    columns, and that the axes are returned in a labelled dictionary instead
+    of an array.  See also `.Figure.subplot_mosaic`.
```

Comment:
```suggestion
    A simple declarative way to create a figure and a grid of axes, that can
    also span multiple rows and columns. The axes are returned in a labelled
    dictionary instead of an array.  See also `.Figure.subplot_mosaic` and
    :doc:`/tutorials/provisional/mosaic`.
```

Reply:
Mostly took this suggestion, though I'm not really sure what you mean by "declarative way" (are their other ways?) so I just said "simple way"....

Reply:
Ok. Simple enough. :)

I'd say that declarative is when you specify parameters for the layout and get back all the Axes at once; vs. in a non declarative way, you call a function (`subplot`, `subplot2grid`) for creating each Axes. But possibly we should not burden the reader with the term "declarative".